protobuf/source_reflection.go

```diff
@@ -83,19 +86,23 @@ func (s *grpcreflectSource) FindMessage(messageType string) (*desc.MessageDescri
 		return nil, nil
 	}
 
+	if err != nil {
+		return nil, wrapReflectionError(err)
+	}
+
 	return msg, err
 }
 
 func (s *grpcreflectSource) FindService(fullyQualifiedName string) (*desc.ServiceDescriptor, error) {
 	service, err := s.client.ResolveService(fullyQualifiedName)
 	if err != nil {
-		available, availableErr := s.client.ListServices()
-		if availableErr != nil {
-			err = availableErr
+		if !grpcreflect.IsElementNotFoundError(err) {
+			return nil, wrapReflectionError(err)
 		}
 
-		if !grpcreflect.IsElementNotFoundError(err) {
-			return nil, err
+		available, availableErr := s.client.ListServices()
+		if availableErr != nil && !grpcreflect.IsElementNotFoundError(availableErr) {
+			return nil, wrapReflectionError(availableErr)
```

Comment:
missing unitests

Reply:
Catching this one is a bit tricky as the grpcReflect client doesn't expose an interface. Also, this a [minor rejig](https://github.com/yarpc/yab/pull/337#discussion_r617695948) of statements from before and this change doesn't reduce the coverage around here. Needs more refactor, which can be done outside of this PR.